relay_messages.go

```diff
@@ -132,6 +135,13 @@ func newLazyCallReq(f *Frame) lazyCallReq {
 	arg1Len := int(binary.BigEndian.Uint16(f.Payload[cur : cur+2]))
 	cur += 2
 	cr.method = f.Payload[cur : cur+arg1Len]
+
+	// arg2~2
+	cur += arg1Len
+	cr.arg2StartOffset = cur + 2
+	cr.arg2EndOffset = cur + 2 /*arg2 len*/ + int(binary.BigEndian.Uint16(f.Payload[cur:cur+2]))
+	// arg2 is fragmented if we don't see arg3 in this frame.
+	cr.isArg2Fragmented = int(cr.Header.FrameSize()) <= (FrameHeaderSize+cr.arg2EndOffset) && f.Payload[_flagsIndex]&hasMoreFragmentsFlag != 0
```

Comment:
instead of using frame size here, if you use `PayloadSize()`, then you won't need to add `FrameHeaderSize` and it'll be a litlte simpler.

similarly, don't access the flags directly, use the method `HasMoreFragments()`

```go
cr.isArg2Fragmented = int(cr.Header.PayloadSize()) <= cr.arg2EndOffset && cr.HasMoreFragments()
```

can we add a test for case where a call doesn't set the has more fragments flag, and doesn't have arg3

Reply:
Good point - I think I made the mistake that the offset is from start of the payload to Arg2, not from the start of frame (which includes header).